platform/android/java/lib/src/org/godotengine/godot/input/GodotGestureHandler.java

```diff
@@ -70,15 +72,17 @@ public void onLongPress(MotionEvent event) {
 		//Log.i("GodotGesture", "onLongPress");
 	}
 
+	@RequiresApi(api = Build.VERSION_CODES.M)
 	@Override
 	public boolean onDoubleTap(MotionEvent event) {
 		//Log.i("GodotGesture", "onDoubleTap");
 		final int x = Math.round(event.getX());
 		final int y = Math.round(event.getY());
+		final int button = event.getSource() == InputDevice.SOURCE_MOUSE ? MotionEvent.BUTTON_PRIMARY : 0; // for now any double click with mouse counts as primery click
```

Comment:
That doesn't sound right. Why do we assume that all double click are coming from the primary button. This logic would break apps/games that try to support double clicks from other mouse buttons.

Reply:
Any event other than `onGenericMotionEvent` doesn't get the actual button being clicked.
fully implementing double click on mouse with real button click needs to be implemented by hand in `onGenericMotionEvent`... For now I decided to just implement it fast with doubletap and assuming primary when using mouse... It does brick double click by right or middle

Reply:
Any event other than `onGenericMotionEvent` doesn't get the actual button being clicked.
fully implementing double click on mouse with real button click needs to be implemented by hand in `onGenericMotionEvent`... For now I decided to just implement it fast with doubletap and assuming primary when using mouse... It does brick double click by right or middle

Reply:
Any event other than `onGenericMotionEvent` doesn't get the actual button being clicked.
fully implementing double click on mouse with real button click needs to be implemented by hand in `onGenericMotionEvent`... For now I decided to just implement it fast with doubletap and assuming primary when using mouse... It does brick double click by right or middle

Reply:
Any event other than `onGenericMotionEvent` doesn't get the actual button being clicked.
fully implementing double click on mouse with real button click needs to be implemented by hand in `onGenericMotionEvent`... For now I decided to just implement it fast with doubletap and assuming primary when using mouse... It does brick double click by right or middle

Reply:
The `GestureDetector` that triggers this callback also has a `onGenericMotionEvent` which is similar to its `onTouchEvent` but for generic motion events.
Is the mouse `double tap` gesture not detected when using that callback?

Reply:
This is under GestureDetector, my problem was that I tried to use `getActionButton` which only works for `ACTION_BUTTON_PRESSED/RELEASE`, changing to `getButtonState` fixed that and after testing it looks like double tapping by multibuttons only sends the primary and not all the buttons pressed